lib/kamal/cli/app.rb

```diff
@@ -42,7 +42,9 @@ def boot
 
               execute *auditor.record("Booted app version #{version}"), verbosity: :debug
 
-              execute *app.run(hostname: "#{host}-#{SecureRandom.hex(6)}")
+              hostname = "#{host}-#{SecureRandom.hex(6)}"
+
+              execute *app.run(hostname: (hostname[-64..-1] || hostname).gsub(/^-+/, ''))
```

Comment:
Can we truncate from the end of the hostname instead? That means that we no longer need to look for `-`s as they are allowed on the end of a hostname. We'll want to strip any trailing `.`s though.

Also I think we should extract a method here to convert `host` to ` hostname.

Reply:
We can truncate from the end of the string, but then we might be truncating the random part that is added in `SecureRandom.hex(6)`, unless you suggest to truncate the host part and keep 12 random characters appended. WDYT?

Reply:
Random part seems less relevant, since you'd mostly identify service by the name, I think.

Reply:
@kryachkov - yes sorry I meant truncate first, then append the random string!

Reply:
> @kryachkov - yes sorry I meant truncate first, then append the random string!

Working on a fix! Thanks

Reply:
I accidentally messed up this PR after rebasing my fork my source, so I had to open a new PR - https://github.com/basecamp/kamal/pull/762